lib/crypto.js

```diff
@@ -584,17 +584,22 @@ ECDH.prototype.generateKeys = function generateKeys(encoding, format) {
 ECDH.prototype.getPublicKey = function getPublicKey(encoding, format) {
   var f;
   if (format) {
-    if (typeof format === 'number')
-      f = format;
-    if (format === 'compressed')
-      f = constants.POINT_CONVERSION_COMPRESSED;
-    else if (format === 'hybrid')
-      f = constants.POINT_CONVERSION_HYBRID;
-    // Default
-    else if (format === 'uncompressed')
-      f = constants.POINT_CONVERSION_UNCOMPRESSED;
-    else
-      throw new TypeError('Bad format: ' + format);
+    switch (format.length) {
+      case 6:
+        if (format === 'hybrid')
```

Comment:
It will go through all the cases in the `if-else-if` chain. The `switch` statement reduces it into 2 checks. Same code as https://github.com/nodejs/node/blob/master/lib/_http_outgoing.js#L513